contextual_test.go

```diff
@@ -56,3 +59,52 @@ func ExampleFlushLogger() {
 	// Output:
 	// flushing...
 }
+
+func BenchmarkPassingLogger(b *testing.B) {
+	b.Run("context", func(b *testing.B) {
+		ctx := klog.NewContext(context.Background(), klog.Background())
+		var finalCtx context.Context
+		for n := b.N; n > 0; n-- {
+			finalCtx = passCtx(ctx)
+		}
+		runtime.KeepAlive(finalCtx)
+	})
+
+	b.Run("directly", func(b *testing.B) {
+		logger := klog.Background()
+		var finalLogger klog.Logger
+		for n := b.N; n > 0; n-- {
+			finalLogger = passLogger(logger)
+		}
+		runtime.KeepAlive(finalLogger)
+	})
+}
+
+func BenchmarkExtractLogger(b *testing.B) {
+	b.Run("context", func(b *testing.B) {
+		ctx := klog.NewContext(context.Background(), klog.Background())
+		var finalLogger klog.Logger
+		for n := b.N; n > 0; n-- {
+			finalLogger = extractCtx(ctx)
+		}
+		runtime.KeepAlive(finalLogger)
+	})
+
+	b.Run("directly", func(b *testing.B) {
```

Comment:
I've changed to "from context" and removed the "directly" here, because it's the same as the case above (there's no additional work for extraction).